pythontest.py

```diff
@@ -1,64 +1,82 @@
 
+def getExtension(file):
+    return os.path.splitext(file)[1]
+
 
 def formatcheck(file):
     # Splitting path and file extension off
+    file_extension = getExtension(file)
     filepath = os.path.dirname(file)
-    file_extension = os.path.splitext(file)[1]
-
     # Changing working directory to filepath
     os.chdir(filepath)
 
     # Defining allowed file extensions
-    allowed_extension = ['.xlsx', '.csv', '.xls']
+    allowed_extensions = ['.xlsx', '.csv', '.xls']
 
     # checking if the path exists
     if not os.path.exists(file):
-        print('The file path for the following file does not exist:', file)
+        # Throw file path error
+        sys.exit('The file path for the following file does not exist ' + file)
 
     # checking if the format is valid
-    if file_extension not in allowed_extension:
-        print('The following file has none supported file format:', file)
-    return file_extension
+    if file_extension not in allowed_extensions:
+        # Throw file format error
+        sys.exit('The following file has none supported file format: ' + file)
 
 formatcheck(file1)
 formatcheck(file2)
 
-
 def reading_file(file):
     filepath = os.path.dirname(file)
     # Changing working directory to filepath
     os.chdir(filepath)
     # Splitting file extension
-    file_extension = os.path.splitext(file)[1]
+    file_extension = getExtension(file)
+
 
     #Read file based on fileformat
-    if file_extension == '.xlsx' or 'xls':
+    if file_extension == '.xlsx' or file_extension == '.xls':
         # Reading excel and storing it into dataframe
         return pd.read_excel(file)
     else:
         # Reading csv and storing it into dataframe
         return pd.read_csv(file)
 
-data1 = reading_file(file1)
-data2 = reading_file(file2)
 
-def header_check(data):
-    #reading for 3 columns of file
-    headers = (data.columns)[:3]
-    #checking headers on correct format
-    if not (headers==(['id', 'name', 'price'])).all():
-        print('The file does not have the right header columns. Please provide format: id, name, price')
+origin = reading_file(file1)
+destination = reading_file(file2)
 
-header_check(data1)
-
-def file_copy(data_file1,data_file2):
-    # Check if id is present in file2
-    print(data_file1[0])
-
-
-file_copy(data1,data2)
\ No newline at end of file
+def headerCheck(data):
+    #reading first 3 columns of file
+    headers = data.columns
+    required_columns = ['id', 'name', 'price']
+    #checking headers on correct format
+    for column in required_columns:
+        if not column in headers:
+            sys.exit('The file does not have the right header columns. Please provide those columns in each file: id, name, price')
+
+headerCheck(origin)
+headerCheck(destination)
+
+def file_copy(origin,destination):
+    # Creating first match criteria via ID
+    # Read all IDs of origin file and converting into regular array
+    origin_ids = origin['id'].to_numpy()
```

Comment:
I don't think you need to convert this to numpy array, it should be fine to work directly on the pandas dataframe